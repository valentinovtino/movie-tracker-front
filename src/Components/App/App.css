@font-face {
  font-family: title;
  src: url(../../Fonts/soup.TTF);
}

.App {
  text-align: center;
  background-color: #222;
}

.App-header {
  background-color: #222;
  height: 223px;
  padding: 20px;
  color: white;
}

.App-title {
  text-align: center;
  letter-spacing: 10px;
  font-size: 55px;
  color: red;
  text-shadow: 3px 0 black, 0 3px black, 1px 0 black, 0 -1px black; 
  font-family: 'title'
}

.nav-container {
  display: flex;
  
}

.nav-btn {
  color: white;
  margin-left: 60px
}


