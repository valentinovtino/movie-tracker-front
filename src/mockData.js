export const mockObj = {
  "id": 28,
  "page": 1,
  "results": [
    {
      "adult": false,
      "backdrop_path": "/bOGkgRGdhrBYJSLpXaxhXVstddV.jpg",
      "genre_ids": [
        12,
        878,
        14,
        28
      ],
      "id": 299536,
      "original_language": "en",
      "original_title": "Avengers: Infinity War",
      "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.",
      "release_date": "2018-04-25",
      "poster_path": "/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg",
      "popularity": 555.264536,
      "title": "Avengers: Infinity War",
      "video": false,
      "vote_average": 8.6,
      "vote_count": 3383
    }
  ]
};

export const mockCleanedData = [{"averageRating": 8.6, "overview": "As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.", "posterPath": "/7WsyChQLEftFiDOVTGkv3hFpyyt.jpg", "releaseData": "2018-04-25", "title": "Avengers: Infinity War"}]