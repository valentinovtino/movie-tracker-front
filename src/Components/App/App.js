import React, { Component } from 'react';
import { fetchMovieData } from '../../apiCalls/api';
import './App.css';
import CardHolderContainer from '../../Containers/CardHolderContainer';
import { NavLink, Route } from 'react-router-dom'
import LogInPage from '../LogInPage/LogInPage.js'
import PropTypes from 'prop-types';

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {

    }
  }

  async componentDidMount() {
    const url = 'https://api.themoviedb.org/3/genre/28/movies?api_key=b896e3605c38d8f6ffb3a181d5bb558d&language=en-US&include_adult=false&sort_by=created_at.asc'

    const response = await fetchMovieData(url);
    const results = response.results

    this.props.storeMovies(results)

  }
  
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <NavLink to="/"><h1 className="App-title">Welcome to Movie-Tracker</h1></NavLink>
          <NavLink to="/login"><span>Log in</span></NavLink>
          <LogInPage />
        </header>
        <Route exact match='/' component={CardHolderContainer} />
      </div>
    );
  }
}



export default App;
