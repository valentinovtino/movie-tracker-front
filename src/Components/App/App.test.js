import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';
import { mockObj } from '../../mockData.js';
describe('App', () => {
  let wrapper;
  let mockUserLoggedOut = jest.fn();
  let mockStoreMovies = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<App user={{}} storeMovies={mockStoreMovies} userLoggedOut={mockUserLoggedOut}/>, {disableLifecycleMethods: true})
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Log Out button if there is a user', () => {
    wrapper = shallow(<App user={{name: 'elvis'}} storeMovies={mockStoreMovies} userLoggedOut={mockUserLoggedOut}/>, {disableLifecycleMethods: true})

    expect(wrapper.find('.log-out-button').length).toEqual(1);
    expect(wrapper.find('.log-in-button').length).toEqual(0);
  });

  it('should render Log In button if there is no user', () => {
    expect(wrapper.find('.log-in-button').length).toEqual(1);
    expect(wrapper.find('.log-out-button').length).toEqual(0);
  });

  it('should call userLoggedOut callback when button is clicked', () => {
    wrapper = shallow(<App user={{name: 'elvis'}} storeMovies={mockStoreMovies} userLoggedOut={mockUserLoggedOut}/>, {disableLifecycleMethods: true})

    wrapper.find('.log-out-button').simulate('click');

    expect(mockUserLoggedOut).toHaveBeenCalled();
  });

  it('should call storeMovies on componentDidMount', async ()=> {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({json: () => Promise.resolve(mockObj)}))

    await wrapper.instance().componentDidMount();

    expect(mockStoreMovies).toHaveBeenCalledWith(mockObj.results);
  });

});









