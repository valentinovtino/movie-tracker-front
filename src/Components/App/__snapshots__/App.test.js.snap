// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    storeMovies={[Function]}
    user={Object {}}
    userLoggedOut={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <h1
                    className="App-title"
          >
                    MOVIE - TRACKER
          </h1>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/login"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/favorites"
          />
</header>,
        <div
          className="nav-container"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav-btn"
                    to="/"
          >
                    Home
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="log-in-button nav-btn"
                    to="/login"
          >
                    Log in
          </NavLink>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"

                    className="log-in-button nav-btn"

                    to="/favorites"
          >
                    Your Favorites
          </NavLink>

</div>,
</header>,
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/login"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/favorites"
/>,

      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="App-title"
>
              MOVIE - TRACKER
</h1>,
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/login"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/favorites"
/>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MOVIE - TRACKER",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "MOVIE - TRACKER",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/favorites",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav-btn"
              to="/"
>
              Home
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="log-in-button nav-btn"
              to="/login"
>
              Log in
</NavLink>,
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="log-in-button nav-btn"

              to="/favorites"
>
              Your Favorites
</NavLink>,
          ],
          "className": "nav-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Home",
              "className": "nav-btn",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Log in",
              "className": "log-in-button nav-btn",
              "to": "/login",
            },
            "ref": null,
            "rendered": "Log in",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Your Favorites",
              "className": "log-in-button nav-btn",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Your Favorites",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Your Favorites",
              "className": "log-in-button",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Your Favorites",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <h1
                        className="App-title"
            >
                        MOVIE - TRACKER
            </h1>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/login"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/favorites"
            />
</header>,
          <div
            className="nav-container"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-btn"
                        to="/"
            >
                        Home
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="log-in-button nav-btn"
                        to="/login"
            >
                        Log in
            </NavLink>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"

                        className="log-in-button nav-btn"

                        to="/favorites"
            >
                        Your Favorites
            </NavLink>

</div>,

</header>,
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/login"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/favorites"
/>,

        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="App-title"
>
                MOVIE - TRACKER
</h1>,
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/login"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/favorites"
/>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MOVIE - TRACKER",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "MOVIE - TRACKER",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/favorites",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-btn"
                to="/"
>
                Home
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="log-in-button nav-btn"
                to="/login"
>
                Log in
</NavLink>,
              <NavLink
                activeClassName="active"
                ariaCurrent="true"

                className="log-in-button nav-btn"

                to="/favorites"
>
                Your Favorites
</NavLink>,
            ],
            "className": "nav-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Home",
                "className": "nav-btn",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Log in",
                "className": "log-in-button nav-btn",
                "to": "/login",
              },
              "ref": null,
              "rendered": "Log in",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Your Favorites",
                "className": "log-in-button nav-btn",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Your Favorites",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Your Favorites",
                "className": "log-in-button",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Your Favorites",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
